tools/tck-api/src/test/scala/org/opencypher/tools/tck/CypherValueParserTest.scala

```diff
@@ -62,6 +62,7 @@ class CypherValueParserTest extends FunSuite with Matchers {
     CypherValue("''") should equal(CypherString(""))
     CypherValue("'-1'") should equal(CypherString("-1"))
     CypherValue("null") should equal(CypherNull)
+    CypherValue("NaN") should equal(CypherNaN)
```

Comment:
@sherfert This is interesting. Are you intending to write TCK scenarios that assert `NaN`s being returned from Cypher? The language doesn't have a literal syntax for it at the moment (although you could mimic it with e.g. `1.0/0.0`). I believe it has been said in the past that Cypher should convert `NaN`s to `null` as a result value, to unify the values.

Reply:
I don't perceive the lack of a literal as problematic for having it in result assertions in TCK tests.

I actually did write these tests in ef32837 . If the expectation is that Cypher converts them to `null`, that is not what neo4j is doing at least. 

Sorry for not waiting on another review and merging pre-maturely. What do you think how we should proceed with `NaN`s?

Reply:
1. I agree, tentatively.

2. Right.

3. I think we should discuss it in the CLG.